qt/applications/workbench/workbench/app/mainwindow.py

```diff
@@ -739,18 +739,23 @@ def open_about(self):
         about = AboutPresenter(self)
         about.show()
 
+    @staticmethod
+    def _get_setting(settings, key, type):
+        """The 'PyQt_PyObject' (1024) type is sometimes used for settings which have an unknown type."""
+        try:
+            return settings.get(key, type=type)
+        except TypeError:
+            value = settings.get(key, type=QVariant.typeToName(1024))
+            return value if isinstance(value, type) else type(*value)
+
     def readSettings(self, settings):
         qapp = QApplication.instance()
 
         # get the saved window geometry
-        window_size = settings.get('MainWindow/size')
-        if not isinstance(window_size, QSize):
-            window_size = QSize(*window_size)
-        window_pos = settings.get('MainWindow/position')
-        if not isinstance(window_pos, QPoint):
-            window_pos = QPoint(*window_pos)
+        window_size = self._get_setting(settings, 'MainWindow/size', type=QSize)
```

Comment:
Just reading over the code and I think we now have a bit of inconsistency in how I should access a setting. Sometimes a straight `get` passing the `type` is used and then here the `_get_setting` wrapper to catch if the value is not of the expected type.
As someone new to the code it's not immediately obvious what I should do here.

Can the behaviour in capturing the possible unknown type be put in the settings class so everything just goes through `get`?